### ISSUE TITLE
make generate_yaml command broken for TAGs with of type int

### ISSUE THREAD

**Issue by GregHanson** [istio/istio#10010]
**Describe the bug**
from the release-1.1 branch:
add `set -x` to start of /install/updateVersion.sh script
run `./install/updateVersion.sh -a "gihanson,111618"`

```
/Users/gihanson/go/out/darwin_amd64/release/helm template --set global.tag=1116 \
		--name=istio \
		--namespace=istio-system \
		--set global.hub=gihanson \
		--set global.proxy.enableCoreDump=false \
		--set istio_cni.enabled=false \
		 \
		--values install/kubernetes/helm/istio/values-istio.yaml \
		install/kubernetes/helm/istio >> install/kubernetes/istio.yaml
Error: render error in "istio/charts/security/templates/create-custom-resources-job.yaml": template: istio/charts/security/templates/create-custom-resources-job.yaml:58:66: executing "istio/charts/security/templates/create-custom-resources-job.yaml" at <32>: wrong type for value; expected string; got float64
make: *** [istio.yaml] Error 1
```

the following line in the yaml does not work for int TAG's
`name: istio-security-post-install-{{ .Values.global.tag | trunc 32 }}`

Following lines cause problems:
https://github.com/istio/istio/blame/release-1.1/install/kubernetes/helm/subcharts/security/templates/create-custom-resources-job.yaml#L58
https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/subcharts/security/templates/cleanup-secrets.yaml#L68

Broken by PR https://github.com/istio/istio/pull/9971

**Comment by tiswanso** [comment]
FWIW, the issue seems to be pretty narrow to specific TAG formats.  I noticed no problems with the following tags:
`TAG=release-1.1-latest-daily`
`TAG=master-latest-daily`
`TAG=f9d12c36c87dfa46187641711a1953ed467d3ea8`
`TAG=f9d12c36c`

**Comment by GregHanson** [comment]
if the TAG is interpreted as string, everything is fine

**Comment by GregHanson** [comment]
resolved by #10021
